edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -91,33 +91,80 @@ public async Task<IList<string>> GetAllIds()
             }
         }
 
```

Comment:
Should we rename this class to ServiceIdentityHierarchy to match IServiceIdentityHierarchy?

Reply:
originally it was IServiceIdentityTree, but we wanted to separate the interface definition from the implementation detail.

For example, the hierarchy can be backed by something other than a tree, as in the case of multiple-parents in the future, it'll be graph.  But interface contracts won't really need to change.

---
In reply to: [458492036](https://github.com/Azure/iotedge/pull/3248#discussion_r458492036) [](ancestors = 458492036)